lib/zendesk_api/collection.rb

```diff
@@ -395,8 +392,12 @@ def set_page_and_count(body)
 
       if cbp_response?(body)
         # We'll delete the one we don't need in #next or #prev
-        @options["page"]["after"] = body["meta"]["after_cursor"]
-        @options["page"]["before"] = body["meta"]["before_cursor"]
+        if @options.page.is_a?(Hash)
```

Comment:
```suggestion
        if cbp_request?
```

This needs an explanation, and the comment above might live better in the next line now.

Reply:
HA! the code is slightly different now... but here's the reason why: 
I found out that two live specs started to fail because even though the code wasn't doing an explicit CBP request, the response was CBP, containing the `has_more` and before/after cursors. One of the cases is `api/v2/trigger_categories.json` . In the previous version of this gem, you could make the request with `client.trigger_categories.fetch` but you would never go past the first page.

Reply:
I wrote a test for this case, it's in a separate commit.